cvmfs/download.h

```diff
@@ -138,6 +145,11 @@ struct JobInfo {
     probe_hosts = false;
     head_request = false;
     follow_redirects = false;
+    pid = -1;
```

Comment:
Ah, that's indeed nasty.  Of course at the moment we do everything we can to enable session reuse.  That requires a careful look into the download manager.  For a start, every curl handle using HTTPS can be encapsulated in a different share handle.  We should use a memory pool of share handles to memory thrashing.  At a later point, we can be smarter and combine easy handles with the same certificate in the same share handle.